node/core/candidate-selection/src/lib.rs

```diff
@@ -126,24 +137,86 @@ impl JobTrait for CandidateSelectionJob {
 	type ToJob = ToJob;
 	type FromJob = FromJob;
 	type Error = Error;
-	type RunArgs = ();
+	type RunArgs = SyncCryptoStorePtr;
 	type Metrics = Metrics;
 
 	const NAME: &'static str = "CandidateSelectionJob";
 
 	/// Run a job for the parent block indicated
 	//
 	// this function is in charge of creating and executing the job's main loop
-	#[tracing::instrument(skip(_relay_parent, _run_args, metrics, receiver, sender), fields(subsystem = LOG_TARGET))]
+	#[tracing::instrument(skip(relay_parent, keystore, metrics, receiver, sender), fields(subsystem = LOG_TARGET))]
 	fn run(
-		_relay_parent: Hash,
-		_run_args: Self::RunArgs,
+		relay_parent: Hash,
+		keystore: Self::RunArgs,
 		metrics: Self::Metrics,
 		receiver: mpsc::Receiver<ToJob>,
-		sender: mpsc::Sender<FromJob>,
+		mut sender: mpsc::Sender<FromJob>,
 	) -> Pin<Box<dyn Future<Output = Result<(), Self::Error>> + Send>> {
 		Box::pin(async move {
-			let job = CandidateSelectionJob::new(metrics, sender, receiver);
+			macro_rules! try_runtime_api {
```

Comment:
Does this macro really needs to be declared in this function?

Reply:
Why? This blows up the function and makes it unreadable. Putting it above will not put it into any broader scope, besides the scope of this module.

If we follow this reasoning, every free standing function that is called in here should be declared in the function that uses it.

Reply:
moved outside